src/System.Management.Automation/engine/Modules/ModuleSpecification.cs

```diff
@@ -266,92 +273,60 @@ public static bool TryParse(string input, out ModuleSpecification result)
-            else if (x != null && y != null)
+            if (x == y)
             {
-                if (x.Name != null && y.Name != null)
-                {
-                    result = x.Name.Equals(y.Name, StringComparison.OrdinalIgnoreCase);
-                }
-                else
-                {
-                    result = true;
-                }
-                if (result)
-                {
-                    if (x.Guid.HasValue && y.Guid.HasValue)
-                    {
-                        result = x.Guid.Equals(y.Guid);
-                    }
-                }
-                if (result)
-                {
-                    if (x.Version != null && y.Version != null)
-                    {
-                        result = x.Version.Equals(y.Version);
-                    }
-                    else if (x.Version != null || y.Version != null)
-                    {
-                        result = false;
-                    }
-
-                    if (x.MaximumVersion != null && y.MaximumVersion != null)
-                    {
-                        result = x.MaximumVersion.Equals(y.MaximumVersion);
-                    }
-                    else if (x.MaximumVersion != null || y.MaximumVersion != null)
-                    {
-                        result = false;
-                    }
-
-                    if (result && x.RequiredVersion != null && y.RequiredVersion != null)
-                    {
-                        result = x.RequiredVersion.Equals(y.RequiredVersion);
-                    }
-                    else if (result && (x.RequiredVersion != null || y.RequiredVersion != null))
-                    {
-                        result = false;
-                    }
-                }
+                return true;
             }
 
-            return result;
+            return x != null && y != null
+                && String.Equals(x.Name, y.Name, StringComparison.OrdinalIgnoreCase)
+                && Guid.Equals(x.Guid, y.Guid)
+                && Version.Equals(x.RequiredVersion, y.RequiredVersion)
+                && Version.Equals(x.Version, y.Version)
+                && String.Equals(x.MaximumVersion, y.MaximumVersion);
         }
 
+        /// <summary>
+        /// Get a property-based hashcode for a ModuleSpecification object.
+        /// </summary>
+        /// <param name="obj">The module specification for the object.</param>
+        /// <returns>A hashcode that is always the same for any module specification with the same properties.</returns>
+        /// <remarks>
+        /// The algorithm in this method is adapted from a StackOverflow answer from Jon Skeet: https://stackoverflow.com/a/263416/9944203
+        /// </remarks>
         public int GetHashCode(ModuleSpecification obj)
         {
-            int result = 0;
+            if (obj == null)
+            {
+                return 0;
+            }
+
+            // Initial prime value
+            int p = 23;
+
```

Comment:
If the only use of p is to initialize `result` why not just initialize `result` directly.

Reply:
I figured it looks better than just throwing `23` in there (I noticed a few stack overflow answers on this tended to say "I dunno why we used that number", so I thought it was worth calling them `p` and `q`). But not attached to it -- can happily change them back to inline constants.